ddl/ddl_api.go

```diff
@@ -1089,6 +1075,10 @@ func setDefaultAndComment(ctx context.Context, col *table.Column, options []*ast
 			col.Flag |= mysql.NotNullFlag
 		case ast.ColumnOptionNull:
 			col.Flag &= ^uint(mysql.NotNullFlag)
+		case ast.ColumnOptionAutoIncrement:
```

Comment:
Currently only allow keeping or dropping auto_increment flag. Set new flag [is not allowed](https://github.com/pingcap/tidb/pull/3748/files#diff-edfa54cbe2fdb418f0a77bd0ec4a56f2R1171).